google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -278,6 +278,8 @@ public Job waitFor(RetryOption... waitOptions) throws InterruptedException {
 
     TableId table = ((QueryJobConfiguration) getConfiguration()).getDestinationTable();
     // TODO(pongad): return QueryResult so we can inject schema.
+    // QueryResponse response = bigquery.getQueryResults(getJobId()); // should return immediately
+    // return new QueryResult(response.getSchema(), response.getTotalRows(), bigquery.listTableData(table));
     return bigquery.listTableData(table);
```

Comment:
Can you call `.withSchema()` here? (I guess not due to pagination?)

Reply:
Right now `listTableData` just returns a `Page` so we don't have a good method to put the schema in.

We can do that once we uncomment the code above it though.